src/Google.Api.Gax/Platform.cs

```diff
@@ -78,21 +105,23 @@ public sealed class Platform
 
         private static async Task<GcePlatformDetails> LoadGceDetails()
         {
-            // TODO: Respect METADATA_EMULATOR_HOST environment variable. Where is this documented?
+            // Check if emulator is in use by looking for an emulator host in environment variable
+            // METADATA_EMULATOR_HOST. This is the undocumented but de-facto mechanism for doing this.
+            var metadataEmulatorHost = Environment.GetEnvironmentVariable("METADATA_EMULATOR_HOST");
             const string metadataUrl = "http://metadata.google.internal/computeMetadata/v1?recursive=true";
             const string metadataFlavorKey = "Metadata-Flavor";
             const string metadataFlavorValue = "Google";
             const long maxContentLength = 512 * 1024; // Maximum allowed metadata size.
             try
             {
-                var httpRequest = new HttpRequestMessage(HttpMethod.Get, metadataUrl);
+                var httpRequest = new HttpRequestMessage(HttpMethod.Get, metadataEmulatorHost ?? metadataUrl);
```

Comment:
Btw, here's how I use this in the metadata library: [here](https://github.com/GoogleCloudPlatform/google-cloud-dotnet/blob/master/apis/Google.Cloud.Metadata.V1/Google.Cloud.Metadata.V1/MetadataClientImpl.cs#L55) and [here](https://github.com/GoogleCloudPlatform/google-cloud-dotnet/blob/master/apis/Google.Cloud.Metadata.V1/Google.Cloud.Metadata.V1/MetadataClientImpl.cs#L55). I think at one point we decided that it was safe to just check for the variable once because its unlikely the presence of the emulator is going to change during the test run of the application.